ekncontent/ekncontent/eknc-query-object.c

```diff
@@ -547,6 +549,19 @@ get_title_clause (EkncQueryObject *self, gchar **terms)
 }
 
 static gchar *
+get_title_clause2 (EkncQueryObject *self, gchar **terms)
+{
+  guint length = g_strv_length (terms);
+  g_auto(GStrv) title_terms = g_new (gchar *, length + 1);
```

Comment:
I just copied a pattern already widely used (in `get_exact_title_clause()`, `get_title_clause()`, `get_body_clause()`, `get_tags_clause()`, and `get_ids_clause()`).

I've pushed another commit which changes all of these to use the pattern you suggest.